torch/csrc/utils/tensor_new.cpp

```diff
@@ -771,16 +773,53 @@ void _validate_sparse_coo_tensor_args(c10::DispatchKey dispatch_key, at::ScalarT
   at::native::_validate_sparse_coo_tensor_args(indices, values, r.intlist(2));
 }
 
+
+
+
+
 Tensor tensor_ctor(c10::DispatchKey dispatch_key, at::ScalarType scalar_type, PyObject* args, PyObject* kwargs) {
+
   static PythonArgParser parser({
     "tensor(PyObject* data, *, ScalarType dtype=None, Device? device=None, bool pin_memory=False, bool requires_grad=False, DimnameList? names=None)",
   });
 
   constexpr int ctor_num_args = 6;
   ParsedArgs<ctor_num_args> parsed_args;
   auto r = parser.parse(args, kwargs, parsed_args);
+
   if (r.idx == 0) {
     PyObject* data = r.pyobject(0);
+
+    PyObject* meta_str = PyUnicode_DecodeFSDefault("meta");
+    PyObject* ellipsis_str = PyUnicode_DecodeFSDefault("Ellipsis");
+
+    PyObject* data_passed = PyObject_Str(data);
+    PyObject* device_passed = r.pyobject(2);
+
+    if(PyObject_RichCompareBool(device_passed, meta_str, Py_EQ)==1 && PyObject_RichCompareBool(data_passed, ellipsis_str, Py_EQ)==1){
```

Comment:
This is a very bad way to do a device equality test. Instead parse the device into a at::Device and then do a normal equality. Ellipsis can be done via object identity test

Reply:
<img width="616" alt="Screenshot 2021-05-13 at 7 38 27 AM" src="https://user-images.githubusercontent.com/13502680/118067457-35a38880-b3be-11eb-9399-55158c51acad.png">

Can I perform checking like this ?

Reply:
<img width="822" alt="Screenshot 2021-05-13 at 7 47 24 AM" src="https://user-images.githubusercontent.com/13502680/118068128-75b73b00-b3bf-11eb-9995-d871f394541f.png">

Can I perform the check in this way ? I'm now using Py_Ellipsis to compare and considering size in the new tensor. Or is there any other method that I can use to perform object identity test ?

Reply:
Equals test is OK because I think ellipsis is a singleton/ String test is not OK.

Reply:
<img width="817" alt="Screenshot 2021-05-13 at 3 23 21 PM" src="https://user-images.githubusercontent.com/13502680/118109696-27c22780-b3ff-11eb-9250-79507cef5865.png">

Will this be fine ? Using `at::Device` to compare as you suggested.

Could you tell me how to post my tests with the PR ? Thank you.

Reply:
<img width="794" alt="Screenshot 2021-05-13 at 3 30 34 PM" src="https://user-images.githubusercontent.com/13502680/118110557-2a714c80-b400-11eb-8243-88ff4c81f652.png">

Using `at::Device` for comparing as you suggested. Would that work ?

Could please tell me how to post my tests with the PR ? Thank you.